### PR TITLE
Peregrine Test Plan added

### PR DESCRIPTION
**Pull request by atharvar28**

### CODE REVIEW

File: test_plans/peregrine/test_plan.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    2. `node : category` filter (description to be added)
+
+    3. `datanode` - Query all custom node types for data fields.
+
+    4. `_node_type` - Query the dictionary structure.
+
+B. Submission Graph
+    
+    1. Link resolver endpoint - This endpoint help to identify if there is a link between two nodes in the dataset. The Link resolver helps us to identify the parent-child relationship between      two nodes.
```
Comment (paulineribeyre) [inline test_plans/peregrine/test_plan.md]:
```suggestion
    1. Link resolver endpoint - This endpoint help to identify if there is a link between two nodes in the dataset. The Link resolver helps us to identify the parent-child relationship between two nodes. 
```

File: test_plans/peregrine/test_plan.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+B. Submission Graph
+    
+    1. Link resolver endpoint - This endpoint help to identify if there is a link between two nodes in the dataset. The Link resolver helps us to identify the parent-child relationship between      two nodes. 
+
+    2. Count resolver endpoint - This endpoint helps us to identify the number of nodes that each node type have. The count resolver is associated with a single node.
+
+C. User Defined Querying
+    
+    1. Generic node - This endpoint provides the user to query with any defined properties in the dictionary for any node in the dictionary in any structure the user wants it to be in. The          user-defined structure should comply with the dictionary rules.
```
Comment (paulineribeyre) [inline test_plans/peregrine/test_plan.md]:
```suggestion
    1. Generic node - This endpoint provides the user to query with any defined properties in the dictionary for any node in the dictionary in any structure the user wants it to be in. The user-defined structure should comply with the dictionary rules. 
```

File: test_plans/peregrine/test_plan.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+             NonExistingNodeProp1
+        }
+      }   
+    ```
+
+    2. `transaction_log` - After submitting metadata with sheepdog, there should be transaction logs containing details (success/failure, entities submitted...).
+
+D. Filters
+    
+    1. `with_path_to`/ `with_path_to_any`/ `without_path_to` - This filter applies to the generic-node node, limiting to the nodes which possess a concrete or any relationship (could include        multiple levels of relationships) to a given node type in the query.
```
Comment (paulineribeyre) [inline test_plans/peregrine/test_plan.md]:
```suggestion
    1. `with_path_to`/ `with_path_to_any`/ `without_path_to` - This filter applies to the generic-node node, limiting to the nodes which possess a concrete or any relationship (could include multiple levels of relationships) to a given node type in the query.
```

File: test_plans/peregrine/test_plan.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+    2. `boolean` - This filter limits the result to only the node instances that have the same boolean property value as given in the filter.
+
+    3. `id` - This filter selects the node of the given UUID.
+
+    4. `ids` - This filter select nodes of the given UUIDs.	
+
+    5. `submitter_id` - This filter select nodes of the given submitter_id.	
+
+    6. `quick_search` - The quick_search filter searches for case insensitive substrings within the node UUID as well as in the unique keys of the JSONB.Currently, for simplicity and               performance, only the `submitter_id` is being used in this filter.
```
Comment (paulineribeyre) [inline test_plans/peregrine/test_plan.md]:
```suggestion
    6. `quick_search` - The quick_search filter searches for case insensitive substrings within the node UUID as well as in the unique keys of the JSONB.Currently, for simplicity and performance, only the `submitter_id` is being used in this filter.
```

File: test_plans/peregrine/test_plan.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    1. Link resolver endpoint - This endpoint help to identify if there is a link between two nodes in the dataset. The Link resolver helps us to identify the parent-child relationship between      two nodes. 
+
+    2. Count resolver endpoint - This endpoint helps us to identify the number of nodes that each node type have. The count resolver is associated with a single node.
+
+C. User Defined Querying
+    
+    1. Generic node - This endpoint provides the user to query with any defined properties in the dictionary for any node in the dictionary in any structure the user wants it to be in. The          user-defined structure should comply with the dictionary rules. 
+      Example : 
+      Correct Structure:
+    ```
```
Comment (paulineribeyre) [inline test_plans/peregrine/test_plan.md]: this is displayed as a code block inside a code block :(

**Comment by atharvar28** [comment]
@paulineribeyre The code block thing works differently each time and various indentations also affect the way it turns out.

File: test_plans/peregrine/test_plan.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        ```    
+        In the above example, the submitter_id for the particular project would be ordered in the ascending order in created_datatime. 
+
+12. `first` - This filter truncates the result set.
+
+13. `offset` - This filter applies offset to the head of the set and displays the result set.
+
+Datasets Endpoints `/datasets`
+```
+false - Need to be logged in to see data. When logged in, only see the projects you have access to.
```
Comment (paulineribeyre) [inline test_plans/peregrine/test_plan.md]: what is true or false is the PUBLIC_DATASETS configuration variable

Comment (paulineribeyre) [reply test_plans/peregrine/test_plan.md]: can you specify that in the test plan?

Comment (atharvar28) [reply test_plans/peregrine/test_plan.md]: @paulineribeyre Done 👍

**Pull request merged**